File: qqbot-module-qqbot-runtime/src/main/java/com/mieai/qqbot/modules/runtime/RobotPluginArtifactHashes.java
package com.mieai.qqbot.modules.runtime;

import java.util.Map;

/** Supplies robot plugin artifact hashes for multi-instance lease admission. */
@FunctionalInterface
public interface RobotPluginArtifactHashes {
    Map<String, String> current();

    static RobotPluginArtifactHashes empty() {
        return Map::of;
    }
}

File: qqbot-module-qqbot-runtime/src/main/kotlin/com/mieai/qqbot/modules/runtime/RobotPluginArtifactHashes.kt
package com.mieai.qqbot.modules.runtime


/** Supplies robot plugin artifact hashes for multi-instance lease admission. */
fun interface RobotPluginArtifactHashes {
    fun current(): Map<String, String>

    companion object {
        fun empty(): RobotPluginArtifactHashes = RobotPluginArtifactHashes { emptyMap() }
    }
}
